pages/linux/pacdiff.md

```diff
@@ -0,0 +1,29 @@
+# pacdiff
+
+> Pacorig, pacnew and pacsave maintenance utility for pacman.
+> If you want to use a different program for diffs, set the `DIFFPROG` enviromental variable to point to your desired binary, with any needed flags. `vim -d` is the default.
```

Comment:
My understanding is that `pacdiff` manages file backups that pacman creates when potentially overwriting/deleting configuration files. Without searching around I wouldn't be able to determine _why_ I might use this. Can we change this description to indicate that `pacdiff` merges and cleans up these backup files?

Reply:
Will do -- passing the `--backup` flag means that the original file is "backed up" if you modify it by overwriting it based on your pacnew.

Reply:
Checked the source to be sure. Relevant code is as follows.
```bash
while :; do
			ask "(V)iew, (M)erge, (S)kip, (R)emove %s, (O)verwrite with %s, (Q)uit: [v/m/s/r/o/q] " "$file_type" "$file_type"
			read c || break
			case $c in
				q|Q) exit 0;;
				r|R) $SUDO rm -v "$pacfile"; break ;;
				o|O)
					if (( BACKUP )); then
						$SUDO mv -v "$file" "$file.bak"
					fi
					$SUDO mv -v "$pacfile" "$file"
					break ;;
				v|V)
					if view_diff "$pacfile" "$file"; then
						break
					fi ;;
				m|M)
					if merge_file "$pacfile" "$file"; then
						break
					fi ;;
				s|S) break ;;
				*) msg2 "Invalid answer." ;;
			esac
		done
```
If overwrite is selected, and *only if overwrite is selected*, passing the backup flag will cause the original file to be stored as a ``.bak``. I was under the misconception that the flag would also backup merges. Editing to clarify the behavior of the flag.